developer_docs/testing.md

```diff
@@ -0,0 +1,472 @@
+- [Writing a test](#Writing-a-test)
+- [Files to be aware of](#Files-to-be-aware-of)
+- [Testing plain components](#Testing-plain-components)
+- [Testing Redux](#Testing-Redux)
+- [How to handle API calls in tests](#How-to-handle-API-calls-in-tests)
+- [Some more background on tests](#Some-more-background-on-tests)
+- [Internationalization](#internationalization)
+- [Tips](#Tips)
+- [More resources](#More-resources)
+- [References](#References)
+
+
+## Testing dependencies
+1. [Jest](https://jestjs.io/)
+2. [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/)
+3. [redux-mock-store](https://github.com/reduxjs/redux-mock-store)
+
+## Useful testing commands
+Run the whole test suite
+```
+npm run test
+```
+
+-----
+
+[Run tests that match the pattern](https://stackoverflow.com/questions/28725955/how-do-i-test-a-single-file-using-jest/28775887). Useful if you're writing one specific test and want to only run that one. 
+```
+npm run test -- someFileName
+```
+
+-----
+Run the tests but update the snapshot if they don't match.
+
+```
+npm run test -- -u
+```
+
+----
+For example, if you wanted to run just the SketchList test but also update the snapshot, you could do:
+```
+npm run test -- Sketchlist.test.js -u
+```
+
+Find more commands in the [Jest documentation](https://jestjs.io/docs/cli).
+
+## Why write tests
+- Good place to start if you're learning the codebase because it's harder to mess up production code
+- Benefits all future contributors by allowing them to check their changes for errors
+- Catches easy-to-miss errors
+- Lets you check your own work and feel more comfortable sumbitting PRs
+- Good practice for large projects
+- Many of the existing components don't have tests yet, and you could write one :-)
+
+## When to run tests
+
+When you make a git commit, the tests will be run automatically for you. 
+
+When you modify an existing component, it's a good idea to run the test suite to make sure it didn't make any changes that break the rest of the application. If they did break some tests, you would either have to fix a bug component or update the tests to match the new expected functionality.
+
+## Writing a test
+Want to get started writing a test for a new file or an existing file, but not sure how?
+### For React components
+(the below assumes we're using proposed folder structure 1)
+1. Make a new file in the ``__tests__`` folder that's directly adjacent to your file. For example, if ``example.jsx`` is in ``src/components``, then you would make a file called ``example.test.jsx`` in ``src/components/__tests__``
+2. Check if the component is connected to redux or not.
+3. If it is, see the redux section below on how to write tests for that.
+4. If it's not, see the section below on writing tests for unconnected components.
+
+### What to test
+For any type of component, you might want to consider testing:
+- User input results in the class's method being called. 
+  ```
```

Comment:
if you change this line to "```js", then this code will be syntax highlighted. same with all of the other instances of code. See [Creating and highlighting code blocks](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks).

Reply:
wow, MAGIC! It worked and looks much better :))